File: csharp/test/Ice/servantLocator/ServerAMD.cs
// Copyright (c) ZeroC, Inc.

using Test;

namespace Ice.servantLocator
{
    namespace AMD
    {
        public class Server : TestHelper
        {
            public override void run(string[] args)
            {
                var initData = new InitializationData();
                initData.properties = createTestProperties(ref args);
                using (var communicator = initialize(initData))
                {
                    communicator.getProperties().setProperty("TestAdapter.Endpoints", getTestEndpoint(0));
                    communicator.getProperties().setProperty("Ice.Warn.Dispatch", "0");

                    var adapter = communicator.createObjectAdapter("TestAdapter");
                    adapter.addServantLocator(new ServantLocatorI("category"), "category");
                    adapter.addServantLocator(new ServantLocatorI(""), "");
                    adapter.add(new TestI(), Ice.Util.stringToIdentity("asm"));
                    adapter.add(new TestActivationI(), Ice.Util.stringToIdentity("test/activation"));
                    adapter.activate();
                    serverReady();
                    adapter.waitForDeactivate();
                }
            }

            public static Task<int> Main(string[] args) =>
                TestDriver.runTestAsync<Server>(args);
        }
    }
}

File: swift/test/Ice/servantLocator/ServerAMD.swift
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import Ice
import TestCommon

class ServerAMD: TestHelperI {
    public override func run(args: [String]) throws {
        let (properties, _) = try createTestProperties(args: args)
        var initData = Ice.InitializationData()
        initData.properties = properties
        initData.classResolverPrefix = "IceServantLocator"
        let communicator = try initialize(initData)
        defer {
            communicator.destroy()
        }
        
        communicator.getProperties().setProperty(key: "TestAdapter.Endpoints",
                                                 value: getTestEndpoint(num: 0))
        communicator.getProperties().setProperty(key: "Ice.Warn.Dispatch", value: "0")
        
        let adapter = try communicator.createObjectAdapter("TestAdapter")
        _ = try adapter.addServantLocator(locator: ServantLocatorI("category", self), category: "category")
        _ = try adapter.addServantLocator(locator: ServantLocatorI("", self), category: "")
        _ = try adapter.add(servant: TestI(), id: Ice.stringToIdentity("asm"))
        _ = try adapter.add(servant: TestActivationI(self), id: Ice.stringToIdentity("test/activation"))
        try adapter.activate()
        serverReady()
        adapter.waitForDeactivate()
    }
}
